src/network/networkd-network.c

```diff
@@ -235,6 +235,18 @@ int network_verify(Network *network) {
                 network->dhcp_use_mtu = false;
         }
 
+        if (network->txqueues > 4096) {
+                log_warning("%s: Invalid TransmitQueues='%d' in [Link] section is set, ignoring.",
+                            network->filename, network->txqueues);
+                network->txqueues = 0;
+        }
+
+        if (network->rxqueues > 4096) {
+                log_warning("%s: Invalid ReceiveQueues='%d' in [Link] section is set, ignoring.",
+                            network->filename, network->rxqueues);
+                network->rxqueues = 0;
+        }
```

Comment:
these checks should be applied when parsing, i.e. in a function `config_parse_rx_tx_queues()` should be added, that refuses == 0 and > 4096.

It's fine to validate/fix up config in network_verify() i guess, but i should probably be used mostly for things that cross multiple settings, i.e. where one setting affects another in some way. As long as the settings are independent I am sure we should already filter out the invalid stuff earlier: when we first parse the data